php/db.php

```diff
@@ -73,6 +76,103 @@ function GetDataForTable($tabName){
 	return $tabData;
 }
 
+function MigrateDatabase() {
+	global $dbConn, $ip, $userAgent, $dbVersion;
+	$migrationsDir = "SQL/versions/";
+
+
+	StartTimer("MigrateDatabase");
+
+	// Check our current database version
+	$sql = "SELECT config_value FROM config WHERE config_key = 'DATABASE_VERSION';";
+	$data = mysqli_query($dbConn, $sql);
+	
+	$config_result = mysqli_fetch_array($data);
+
+	if ($config_result == NULL) {
+		die("Unable to determine the database version. Please add the config value DATABASE_VERSION before running");
+	}
+
+	$currentDatabaseVersion = intval($config_result[0], 10);
+
+	// Check and see if we need to migrate.
+	if ($dbVersion < $currentDatabaseVersion) {
+		die('Database version is out of date with code version');
+	} elseif ($dbVersion == $currentDatabaseVersion) {
+		StopTimer("MigrateDatabase");
+		return;
+	}
+	
+	// Calculate what migrations exist
+	$migrations = array();
+	
+	$migrationDirFiles = scandir($migrationsDir);
+	foreach ($migrationDirFiles as $i => $migrationFile) {
+		// Get the version ID from the filename. Files need to be formatted <versionID>_<randomName>.sql
+		preg_match('/(\d+)_.*\.sql$/', $migrationFile, $matches);
+		if (isset($matches[1])) {
+			$migrationID = intval($matches[1], 10);
+			$migrations[$migrationID] = $migrationFile;
+		}
+	}
+
+	// Go through all of the needed migrations and run them
+	for($i = $currentDatabaseVersion+1; $i <= $dbVersion; $i++) {
+
+		// Make sure that migration ID actually exists. This is so we can have version 10, 11, 14, (notice the missing 13?)
+		if (isset($migrations[$i])) {
+			$migrationScript = $migrations[$i];
+
+			// Update our adminlog so we can see what happened before the migration
+			$escapedIP = mysqli_real_escape_string($dbConn, $ip);
+			$escapedUserAgent = mysqli_real_escape_string($dbConn, $userAgent);
+			$escapedLog = mysqli_real_escape_string($dbConn, "Running $migrationScript to bring DATABASE_VERSION up to $i");
+			$sql = "
```

Comment:
I think this functionality exists in AdminLogData.AddToAdminLog(). If that can't be accessed here, it may be worth making it static in AdminLogData or something? If it's easy to fix, that's great, if not, we can certainly leave it for later

Reply:
Yeah, so this has to be done before any of the models are loaded. Changing AddToAdminLog to static would be a good idea. I'll investigate this more.

I believe there were some concerns about how to handle any possible migrations involving that function and the admin log table. But I'm honestly not sure that there is a proper way of handling that.